independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/core/strategy/SmartPackageFileStrategyHandler.java

```diff
@@ -0,0 +1,48 @@
+package io.quarkus.devtools.codestarts.core.strategy;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import io.quarkus.devtools.codestarts.CodestartStructureException;
+import io.quarkus.devtools.codestarts.core.reader.TargetFile;
+import io.quarkus.devtools.codestarts.utils.NestedMaps;
+
+final class SmartPackageFileStrategyHandler implements CodestartFileStrategyHandler {
+
+    @Override
+    public String name() {
+        return "smart-package";
+    }
+
+    @Override
+    public void process(Path targetDirectory, String relativePath, List<TargetFile> codestartFiles, Map<String, Object> data)
+            throws IOException {
+        checkNotEmptyCodestartFiles(codestartFiles);
+
+        final Optional<String> optPackageName = NestedMaps.getValue(data, "project.package-name");
+        Path targetPath;
+        String content = codestartFiles.get(0).getContent();
+        if (optPackageName.isPresent()) {
+            final String packageName = optPackageName.get();
+            final String packageNameAsDir = packageName.replace(".", "/");
+            targetPath = targetDirectory.resolve(relativePath.replace("org/acme", packageNameAsDir));
+            content = content
+                .replace("import org.acme", "import " + packageName)
```

Comment:
minor thing but I would extract the "org.acme" and "org/acme" into a constant to make it more clear its a special "marker"

Reply:
Sure, this is a draft still :)